app/server/lib/AppEndpoint.ts

```diff
@@ -73,7 +73,10 @@ export function attachAppEndpoint(options: AttachOptions): void {
     if (!docStatus) {
       return res.status(500).json({error: 'no worker'});
     }
-    res.json({docWorkerUrl: customizeDocWorkerUrl(docStatus.docWorker.publicUrl, req)});
+    res.json({
+      docWorkerUrl: customizeDocWorkerUrl(docStatus.docWorker.publicUrl, req),
+      internalDocWorkerUrl: docStatus.docWorker.internalUrl
+    });
```

Comment:
Sounds a good idea. Also would give a hint about who is the caller in the logs.

Maybe then:
 - `/api/worker/:assignmentId/public` 
 - and `/api/worker/:assignmentId/internal`

But there is this `/?*` in the existing path that puzzles me a bit…

Reply:
It's an interesting API design problem.

On the one hand, this sound like an API you'd want exposed only to internal callers, which would call for introducing a prefix (so that e.g. these routes could be blocked on the reverse proxy), as in `/internalapi/worker/:assignmentId`. On the other hand, I can imagine scenarios where an internal call would be made to obtain an external URL, e.g. for redirecting a client to another worker. So, just because it is an internal call doesn't mean you want (only) the internal URL. So you'd end up with `/internalapi/worker/:assignmentId/internal` I guess. Or maybe just have `/internalapi/worker/:assignmentId` return both internal and public URLs as the current endpoint does.

Alternately, keep a single (public) endpoint but include the internal URL only if the caller was detected as internal (perhaps through a header that the reverse proxy blocks).

Or, as I suggested above, maybe drop the need for internal calls to this endpoint altogether?

Reply:
Should be resolved with the latest changes.